src/code/z_skin_matrix.c

```diff
@@ -1,42 +1,728 @@
+    cy = mfB->yz;
+    cz = mfB->zz;
+    cw = mfB->wz;
+    //--------
+    rx = mfA->xx;
+    ry = mfA->xy;
+    rz = mfA->xz;
+    rw = mfA->xw;
+    dest->xz = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_matrix/func_800A7C20.s")
+    rx = mfA->yx;
+    ry = mfA->yy;
+    rz = mfA->yz;
+    rw = mfA->yw;
+    dest->yz = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_matrix/func_800A7C60.s")
+    rx = mfA->zx;
+    ry = mfA->zy;
+    rz = mfA->zz;
+    rw = mfA->zw;
+    dest->zz = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_matrix/func_800A7E70.s")
+    rx = mfA->wx;
+    ry = mfA->wy;
+    rz = mfA->wz;
+    rw = mfA->ww;
+    dest->wz = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
 
+    //---COL4---
+    cx = mfB->xw;
+    cy = mfB->yw;
+    cz = mfB->zw;
+    cw = mfB->ww;
+    //--------
+    rx = mfA->xx;
+    ry = mfA->xy;
+    rz = mfA->xz;
+    rw = mfA->xw;
+    dest->xw = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
+
+    rx = mfA->yx;
+    ry = mfA->yy;
+    rz = mfA->yz;
+    rw = mfA->yw;
+    dest->yw = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
+
+    rx = mfA->zx;
+    ry = mfA->zy;
+    rz = mfA->zz;
+    rw = mfA->zw;
+    dest->zw = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
+
+    rx = mfA->wx;
+    ry = mfA->wy;
+    rz = mfA->wz;
+    rw = mfA->ww;
+    dest->ww = (cx * rx) + (cy * ry) + (cz * rz) + (cw * rw);
+}
+
+/**
+ * "Clear" in this file means the identity matrix.
+ */
+void SkinMatrix_GetClear(MtxF** mfp) {
+    *mfp = &sMtxFClear;
+}
+
+void SkinMatrix_Clear(MtxF* mf) {
+    mf->xx = 1.0f;
+    mf->yy = 1.0f;
+    mf->zz = 1.0f;
+    mf->ww = 1.0f;
+    mf->xy = 0.0f;
+    mf->xz = 0.0f;
+    mf->xw = 0.0f;
+    mf->yx = 0.0f;
+    mf->yz = 0.0f;
+    mf->yw = 0.0f;
+    mf->zx = 0.0f;
+    mf->zy = 0.0f;
+    mf->zw = 0.0f;
+    mf->wx = 0.0f;
+    mf->wy = 0.0f;
+    mf->wz = 0.0f;
+}
+
+void SkinMatrix_MtxFCopy(MtxF* src, MtxF* dest) {
+    dest->xx = src->xx;
+    dest->xy = src->xy;
+    dest->xz = src->xz;
+    dest->xw = src->xw;
+    dest->yx = src->yx;
+    dest->yy = src->yy;
+    dest->yz = src->yz;
+    dest->yw = src->yw;
+    dest->zx = src->zx;
+    dest->zy = src->zy;
+    dest->zz = src->zz;
+    dest->zw = src->zw;
+    dest->wx = src->wx;
+    dest->wy = src->wy;
+    dest->wz = src->wz;
+    dest->ww = src->ww;
+}
+
+/**
+ * Inverts a matrix using a slight modification of the Gauss-Jordan method
+ * (column operations instead of row operations).
+ * returns 0 if successfully inverted
+ * returns 2 if matrix non-invertible (0 determinant)
+ */
+#ifdef NON_MATCHING
+// Saved register usage differences and probably regalloc
+s32 SkinMatrix_Invert(MtxF* src, MtxF* dest) {
+    MtxF mfCopy;
+    s32 i;
+    s32 k;
+    f32 temp1;
+    f32 temp2;
+    f32 diag_element;
+    s32 this_row;
+    s32 this_col;
```

Comment:
not sure if this was mentioned in a previous comment and missed, or if this was overlooked in general, but there should be no snake_case variables. if theres anymore in the file change those too please :D